src/commands/randomViewer.ts

```diff
@@ -1,19 +1,55 @@
+import { differenceInMinutes } from 'date-fns';
 import { ICommandProps } from '.'
 import { client } from '../index'
-import { getViewerList, randomNumber } from "../utils";
+import { getViewerList, isStreamer, randomNumber } from "../utils";
 
+interface ICacheProperties {
+    lastCalled: Date,
+    chosenOne: String
+}
+
+type Cache = Record<string, ICacheProperties>
+
+let cache: Cache = {
+    
+}
+
+// Cooldown in minutes
+const cooldown = 180
 
 export const description = 
-    'Use this command to randomly pick a viewer in the chat and make them feel special SeemsGood'
-export default async({channel}: ICommandProps) => {
-    // Remove # from channel name
+    'Use this command every 3 hours to randomly pick a viewer in the chat and make someone feel special SeemsGood This command has no cooldown for the streamer cus they\'re op PogChamp'
+export default async({channel,context}: ICommandProps) => {
+    // Remove # from "channel"
     let channelName = channel.substring(1)
 
     // Get array of current viewers in chat 
     let viewers = await getViewerList(channelName)
```

Comment:
You should move this api call and the isNan check inside of the if statement. The reason being that we only wan't to do this api call when the cooldown is over. Otherwise we won't even use the viewers variable, so in that case this api call was made for nothing.

Reply:
Nice observations! thanks for the feedback, I see what u mean.